src/ray/core_worker/task_manager.cc

```diff
@@ -640,6 +640,19 @@ void TaskManager::CompletePendingTask(const TaskID &task_id,
       for (const auto &dynamic_return_id : dynamic_returns_in_plasma) {
         it->second.reconstructable_return_ids.insert(dynamic_return_id);
       }
+
+      // Set the number of return values from a streaming generator when it is the
```

Comment:
This has to be called only upon the first execution (here, first execution = false means the task has succeeded at least once, and it is lineage reconstruction). Different number of returns in the second run should be handled by the code below